include/deal.II/base/vectorization.h

```diff
@@ -2707,6 +2707,25 @@ vectorized_transpose_and_store(const bool                    add_into,
 
 
 /**
+ * Relational operator == for VectorizedArray
+ *
+ * @relates VectorizedArray
+ */
+template <typename Number>
+inline DEAL_II_ALWAYS_INLINE
+bool
+operator == (const VectorizedArray<Number> &lhs,
```

Comment:
Actually, this caused a compilation in the main library (exposed at about 97% build completion :-/). 

Here is one such error message that this fixes / works around:
```
In file included from <>/dealii/source/matrix_free/matrix_free.cc:17:
In file included from <>/include/deal.II/matrix_free/matrix_free.templates.h:22:
In file included from <>/include/deal.II/base/tensor_product_polynomials.h:22:
<>/include/deal.II/base/tensor.h:946:13: error: invalid operands to binary expression ('const dealii::VectorizedArray<double>' and 'VectorizedArray<double>')
  Assert (d == internal::NumberType<Number>::value(0.0),
          ~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<>/include/deal.II/base/exceptions.h:330:11: note: expanded from macro 'Assert'
    if (!(cond))                                                              \
          ^~~~
<>/include/deal.II/base/tensor.h:618:10: note: in instantiation of member function 'dealii::Tensor<1, 1, dealii::VectorizedArray<double> >::operator=' requested here
      tmp=internal::NumberType<VectorizedArray<T> >::value(t);
         ^
<>/include/deal.II/base/tensor.h:631:41: note: in instantiation of member function 'dealii::internal::NumberType<dealii::Tensor<1, 1, dealii::VectorizedArray<double> > >::value' requested here
  : value(internal::NumberType<Number>::value(0.0))
                                        ^
<>/include/deal.II/base/tensor.h:376:25: note: in instantiation of member function 'dealii::Tensor<0, 1, dealii::Tensor<1, 1, dealii::VectorizedArray<double> > >::Tensor' requested here
  DEAL_II_CUDA_HOST_DEV Tensor ();
                        ^
<>/include/deal.II/matrix_free/mapping_info.templates.h:511:101: note: in instantiation of member function 'dealii::Tensor<1, 1, dealii::Tensor<1, 1, dealii::VectorizedArray<double> > >::Tensor'
      requested here
                          Tensor<1,(dim>1?dim*(dim-1)/2:1),Tensor<1,dim,VectorizedArray<Number> > > grad_upper;
                                                                                                    ^
<>/deal.II/base/aligned_vector.h:917:6: note: candidate template ignored: could not match 'AlignedVector' against 'VectorizedArray'
bool operator == (const AlignedVector<T> &lhs,
```
I'm not sure why the compiler could evaluate `Assert (d == Number(),...)` before but not `Assert (d == internal::NumberType<Number>::value(0.0),...)`.